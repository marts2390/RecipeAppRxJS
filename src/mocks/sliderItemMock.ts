const sliderItemMock = [
  {
    idCategory: '1',
    strCategory: 'Beef',
    strCategoryDescription: 'Beef is the culinary name for meat from cattle, particularly skeletal muscle. Humans have been eating beef since prehistoric times.[1] Beef is a source of high-quality protein and essential nutrients.',
    strCategoryThumb: 'https://www.themealdb.com/images/category/beef.png',
  },
  {
    idCategory: '2',
    strCategory: 'Chicken',
    strCategoryDescription: 'Chicken is a type of domesticated fowl, a subspecies of the red junglefowl. It is one of the most common and widespread domestic animals, with a total population of more than 19 billion as of 2011.[1] Humans commonly keep chickens as a source of food (consuming both their meat and eggs) and, more rarely, as pets.',
    strCategoryThumb: 'https://www.themealdb.com/images/category/chicken.png',
  },
  {
    idCategory: '3',
    strCategory: 'Dessert',
    strCategoryDescription: 'Dessert is a course that concludes a meal. The course usually consists of sweet foods, such as confections dishes or fruit, and possibly a beverage such as dessert wine or liqueur, however in the United States it may include coffee, cheeses, nuts, or other savory items regarded as a separate course elsewhere. In some parts of the world, such as much of central and western Africa, and most parts of China, there is no tradition of a dessert course to conclude a meal.\r\n\r\nThe term dessert can apply to many confections, such as biscuits, cakes, cookies, custards, gelatins, ice creams, pastries, pies, puddings, and sweet soups, and tarts. Fruit is also commonly found in dessert courses because of its naturally occurring sweetness. Some cultures sweeten foods that are more commonly savory to create desserts.',
    strCategoryThumb: 'https://www.themealdb.com/images/category/dessert.png',
  }
]

export default sliderItemMock
