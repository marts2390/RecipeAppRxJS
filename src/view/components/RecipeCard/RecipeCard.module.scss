.cardContainer {
  &:hover {
    .mealImage {
      transform: scale(1.1);
    }
  }

  .mealImage {
    transition: 0.3s ease;
    height: 15em;
    overflow: hidden;
  }

  .cardContent {
    display: flex;
    align-items: center;
    min-height: 5em;
  }
}
