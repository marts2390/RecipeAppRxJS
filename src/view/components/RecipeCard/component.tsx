import React, { FC } from 'react'
import {
  Card, CardMedia, CardContent, CardActionArea, Fade,
} from '@material-ui/core'
import { Link } from 'react-router-dom'

import { IMealsDataModel } from 'models/mealsDataModel'
import styles from './RecipeCard.module.scss'

interface IRecipeCardComponentProps {
  item: IMealsDataModel
}

const RecipeCardComponent:FC<IRecipeCardComponentProps> = ({ item }) => (
  <Link to={ `/recipes/${item.idMeal}` }>
    <Fade
      in
      timeout={ 500 }
    >
      <Card className={ styles.cardContainer }>
        <CardActionArea>
          <CardMedia
            className={ styles.mealImage }
            image={ item.strMealThumb }
            title={ item.strMeal }
          />
          <CardContent className={ styles.cardContent }>
            <p>{item.strMeal}</p>
          </CardContent>
        </CardActionArea>
      </Card>
    </Fade>
  </Link>
)

export default RecipeCardComponent
