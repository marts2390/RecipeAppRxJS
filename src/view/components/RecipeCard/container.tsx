import React, { FC } from 'react'

import { IMealsDataModel } from 'models/mealsDataModel'

import RecipeCardComponent from './component'

interface IRecipeCardContainerProps {
  item: IMealsDataModel
}

const RecipeCardContainer:FC<IRecipeCardContainerProps> = ({ item }) => (
  <RecipeCardComponent
    item={ item }
  />
)

export default RecipeCardContainer
