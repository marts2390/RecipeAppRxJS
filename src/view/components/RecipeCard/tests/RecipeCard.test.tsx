import React from 'react'
import { shallow } from 'enzyme'
import sliderItemMock from '../../../../mocks/mealItemMock'
import RecipeCard from '../component'

let wrapper
let props

beforeEach(() => {
  props = { item: sliderItemMock }
  wrapper = shallow(<RecipeCard { ...props } />)
})

test('should render recipeCard', () => {
  expect(wrapper).toMatchSnapshot()
})

test('Link should have be correct', () => {
  const link = wrapper.find('Link')
  expect(link.prop('to')).toBe(`/recipes/${props.item.idMeal}`)
})
