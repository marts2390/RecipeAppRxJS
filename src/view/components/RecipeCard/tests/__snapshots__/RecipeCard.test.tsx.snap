// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render recipeCard 1`] = `
<Link
  to="/recipes/52874"
>
  <ForwardRef(Fade)
    in={true}
    timeout={500}
  >
    <WithStyles(ForwardRef(Card))
      className="cardContainer"
    >
      <WithStyles(ForwardRef(CardActionArea))>
        <WithStyles(ForwardRef(CardMedia))
          className="mealImage"
          image="https://www.themealdb.com/images/media/meals/sytuqu1511553755.jpg"
          title="Beef and Mustard Pie"
        />
        <WithStyles(ForwardRef(CardContent))
          className="cardContent"
        >
          <p>
            Beef and Mustard Pie
          </p>
        </WithStyles(ForwardRef(CardContent))>
      </WithStyles(ForwardRef(CardActionArea))>
    </WithStyles(ForwardRef(Card))>
  </ForwardRef(Fade)>
</Link>
`;
